modules/opengl/src/texture/texture2d.cpp

```diff
@@ -117,6 +120,12 @@ void Texture2D::initialize(const void* data) {
                  static_cast<GLsizei>(dimensions_.y), 0, format_, dataType_, data);
     LGL_ERROR;
     forEachObserver([](TextureObserver* o) { o->notifyAfterTextureInitialization(); });
+
+    auto old = resource::remove(resource::GL{id_});
+    resource::add(resource::GL{id_}, Resource{.dims = glm::size4_t{dimensions_, 0, 0},
```

Comment:
won't this add the same resource multiple times if `initialize()` is called more than once?

Reply:
This applies to all textures.

Reply:
If the id is already added it will just update/replace that one.